### DISCUSSION TITLE
Should the Code Editor be an overlay or stay as is?

### DISCUSSION THREAD

**Discussion by kahboom** [KaotoIO/kaoto-ui#267] [General]
If we keep the code editor as is (as part of the PatternFly Grid Layout), then when you collapse/expand it, it shifts the visualization canvas over, which users might find annoying. However, you wouldn't obstruct the zoom buttons on the bottom left of the canvas, which is likely to happen with the overlay (using PatternFly Drawer). One option is to remove those buttons if nobody is using them at all, or to make the code editor a bit shorter.

For reference and additional information, see #217 .
cc @almen-ng

**Comment by almen-ng** [comment]
Another option could be having it permanent on the right and have it as two tabs! 
![Untitled](https://user-images.githubusercontent.com/68764826/161874161-74ba489e-3a43-49f1-b66e-f4a16eec5c6e.png)

**Reply by almen-ng** [reply]
This way there won't be any expansion or collapse of the visualization portion. It can also be resizable to ensure that it is the appropriate size for the users. 

I do not think the step catalog will be used simultaneously with the yaml

**Reply by kahboom** [reply]
Thanks @almen-ng , this sounds like a good idea. Rather than have them both as overlays we can just have them both as panels with tabs. This would also mean that we could remove the `+` button from the right-hand side, since the catalog would always be in the tab.

The code/YAML editor has buttons at the top though that are specific to the editor, so we should only show them when you're on the editor tab. It might look a little more clunky though since we will have the tabs and then the buttons below it. This might be easier to show on a call though. :) I can create an issue to reassess the layout.
